ResearchKit/Common/ORKPlacemark.m

```diff
@@ -0,0 +1,84 @@
+ 
+ Redistribution and use in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKPlacemark.h"
+
+@implementation ORKPlacemark
+
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate addressDictionary:(NSDictionary<NSString *,id> *)addressDictionary {
+    if (!CLLocationCoordinate2DIsValid(coordinate)) {
+        return nil;
+    }
+    self = [super initWithCoordinate:coordinate addressDictionary:addressDictionary];
+    return self;
+}
+
+- (instancetype)initWithPlacemark:(CLPlacemark *)placemark {
+    if (!CLLocationCoordinate2DIsValid(placemark.location.coordinate)) {
+        return nil;
+    }
+    self = [super initWithPlacemark:placemark];
+    return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (!CLLocationCoordinate2DIsValid(self.location.coordinate)) {
+        return nil;
+    }
+    return self;
+}
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (id)copyWithZone:(NSZone *)zone {
+    return [[ORKPlacemark alloc] initWithPlacemark:self];
+}
+
+- (BOOL)isEqual:(id)object {
+    if ([[object class] isSubclassOfClass:[ORKPlacemark class]]) {
+        return (self.location.coordinate.latitude == ((ORKPlacemark *)object).location.coordinate.latitude &&
+                self.location.coordinate.longitude == ((ORKPlacemark *)object).location.coordinate.longitude);
+    } else {
+        return NO;
+    }
+}
```

Comment:
Removing isEqual: causes the equality tests to fail. Since MKPlacemark's implementation of isEqual: returns nil;